admin/src/js/controllers/targets-edit.js

```diff
@@ -95,6 +95,18 @@ angular.module('controllers').controller('TargetsEditCtrl',
         });
     };
 
+    const isTargetUnique = (settings) => {
+      if ($scope.editing) {
+        // Here we are editing a target, assume it is unique 
+        return $q.resolve({ settings: settings, unique: true });
+      }
+      const  items = (settings.tasks && settings.tasks.targets &&
+        settings.tasks.targets.items) || [];
+      const filteredItems = items.filter(item => item.id === $scope.target.id);
+      const unique = filteredItems.length === 0 ? true : false;
```

Comment:
Use [array.some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) here to make this a bit cleaner (and faster!), eg:

```js
const unique = items.some(item => item.id === $scope.target.id);
```